crates/proof-of-sql/src/base/scalar/mod.rs

```diff
@@ -20,6 +20,16 @@ mod commitment_utility;
 pub use commitment_utility::compute_commitment_for_testing;
 use num_bigint::BigInt;
 
+/// Alternative ordering on finite sets considered to be signed.
+pub trait SignedOrd: Ord {
```

Comment:
I'm trying to think if a new trait is the right way to go here, or if it's a new type. For example

```rust
pub struct SignedScalar<S>(S) where S: Scalar;

// implement various num_traits and operations for `SignedScalar`, including `Ord`

// use `SignedScalar` in Decimal75 columns instead of plain Scalar
```

I think something like that is a bigger lift, but I think it has a couple benefits. Namely,
1. The new type makes it very clear what the meaning of the underlying data is, whereas a new trait is sort of agnostic about this (it could be either)
2. Allows for using the `SignedScalar` in generic code w/ trait bounds like `T: Ord` without unexpected results (the type is very clear about which order implementation is the right one)

The drawbacks would be, again, it's a bigger lift. Another one I can think of is that it might be hard to understand which one is the correct one to use elsewhere in the code base, or as we develop new code. I think for this purpose it would be good if `SignedScalar` didn't actually implement `Scalar`, so that we don't make the mistake of using it in the proving protocols, but it could implement `Into<S> where S: Scalar`. I don't know off the top of my head if this thought is compatible with all of our current code.

Reply:
I think the essence of the problem is that any `Scalar` just like bytes and vectors of bytes have both signed and unsigned forms of comparison.